### PR TITLE
Update Automation Framework job specifications

### PR DESCRIPTION
**Pull request by DragosStefanNecula**
## Overview
This PR:
https://github.com/zaproxy/zap-extensions/pull/4326/files

That was part of a effort to close this issue:
https://github.com/zaproxy/zaproxy/issues/7663

Resulted in the documentation being out of date.

This PR addresses that.

Thank you very much and I hope you have a nice day.

## Related Issues
Original PR that was created: https://github.com/zaproxy/zaproxy-website/pull/2859

## Checklist
- [X] Update help
- [X] Update changelog
- [ ] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [X] Sign-off commits
- [X] Squash commits
- [X] Use a descriptive title

### CODE REVIEW

**Comment by DragosStefanNecula** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by DragosStefanNecula** [comment]
If any changes are requested I will get back to you as soon as I can, probably within 24 hours

File: addOns/automation/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 ### Added
+- Updated automation framework documentation and YAML files for `activeScan` job to reflect PR #4326
```
Comment (thc202) [inline addOns/automation/CHANGELOG.md:8]: This should be in Changed or even Fixed not Added, the PR reference is unnecessary. "YAML files", should be "templates"

**Comment by DragosStefanNecula** [comment]
I have actioned your requests and also squashed the commits into one. Thank you so much for your help! Let me know if there's anything else I can do!

**Comment by thc202** [comment]
There are some tests failing (expected templates need to be updated to reflect the changes).

**Comment by DragosStefanNecula** [comment]
I have found 2 template files that I missed to update, and I did check the tests and it does seem it could be significant, hopefully this commit fixes it but I am going to look into it for a bit more

**Comment by DragosStefanNecula** [comment]
I think I need to run ./gradlew spotlessAppl even if the changes are minor, I will try to do that

**Comment by DragosStefanNecula** [comment]
Seems like tests have passed I think

**Review by kingthorin — APPROVED** [review]
LGTM

**Comment by DragosStefanNecula** [comment]
(assuming the PR might close soon): thank you so much to the both of you! this was one of my first few open source contributions and your pointers were helpful :)

**Comment by kingthorin** [comment]
No problem. Great to see new contributors!

**Review by psiinon — APPROVED** [review]

File: addOns/spider/src/main/javahelp/org/zaproxy/addon/spider/resources/help/contents/automation.html [hunk 2]
```diff
@@ -43,7 +43,7 @@ <H2>YAML</H2>
       postForm:                        # Bool: Whether the spider will submit POST forms, default: true
       processForm:                     # Bool: Whether the spider will process forms, default: true
       sendRefererHeader:               # Bool: Whether the spider will send the referer header, default: true
-      threadCount:                     # Int: The number of spider threads, default: 2
+      threadCount:                     # Int: The number of spider threads, default: 2 * Number of available processor cores
```
Comment (thc202) [inline addOns/spider/src/main/javahelp/org/zaproxy/addon/spider/resources/help/contents/automation.html:46]: This is not correct, either revert or change the value in the job parameters (I think the latter is better for consistency).

Comment (DragosStefanNecula) [reply addOns/spider/src/main/javahelp/org/zaproxy/addon/spider/resources/help/contents/automation.html:46]: Hey, there is a few places where Constants.getDefaultThreadCount() gets used within the spiderJob

image
image
image

Some of the tests also intercept getDefaultThreadCount() which I take as further evidence that this is used to set the default:
image

Everywhere that I look, that method seems to be used, could you please help me understand why

Comment (DragosStefanNecula) [reply addOns/spider/src/main/javahelp/org/zaproxy/addon/spider/resources/help/contents/automation.html:46]: I think I did find what you are referring to:

image

File: addOns/spider/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Unreleased
 ### Changed
 - Updated automation framework documentation and templates for `spider` job to reflect changes to the default value of threadCount parameter
+- Updated default threadCount for `spider` job in automation framework to be consistent with other default values for threads
```
[the remark this answers is not in the document]
Comment (DragosStefanNecula) [reply addOns/spider/CHANGELOG.md]: Done!

**Comment by DragosStefanNecula** [comment]
I have looked through the tests and I don't think there is anything that will fail (I can't know for sure because I can't run tests locally and I was unable to troubleshoot it)

**Comment by thc202** [comment]
Thank you!

**Pull request merged**